lib/etso/adapter/types/map.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Etso.Ecto.MapType do
```

Comment:
I think this responsibility of sanitising data should belong to the client, where this type should be implemented in each application and incorporated in the appropriate schema. Etso itself should not munge the data. What I put in should be what I get out.

In other words everything should be 100% exactly the same in a save/load roundtrip via Etso.

Reply:
The reason I had to do this is that most times users persist maps with atom keys and the way other adapters work in this case is to serialize those atoms to their string equivalent. so it's appropriate the same behavior is implemented in etso else we will have users having to keep track of different adapters behavior in situations like this.

Reply:
I think it is not appropriate for Etso to impose this overhead as the idea of ETS is being able to store anything. If customers wanted to serialise / deserialise they would do it themselves. There is an overhead on sanitisation and fundamentally the reason why other DBs would sanitise is because they do not have the concept of atoms… But we are using ETS which is native to Erlang so all types are supported even functions.